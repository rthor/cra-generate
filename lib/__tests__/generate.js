'use strict'

const generate = require('../generate')

jest.mock('fs')
jest.mock('mkpath')

describe('generate', () => {
  test('returns an array of files', () => {
    const data = generate('test-component')
    expect(data.componentName).toBe('TestComponent')
    expect(Array.isArray(data.files)).toBeTruthy()
  })
})
