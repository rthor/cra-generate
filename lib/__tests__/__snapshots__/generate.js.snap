// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate should add semi colons 1`] = `
Array [
  ".TestComponent {}
",
  "export { default } from './TestComponent';
",
  "import React from 'react';
import ReactDOM from 'react-dom';
import TestComponent from './TestComponent';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TestComponent />, div);
});
",
  "import React, { Component } from 'react';
import './TestComponent.css';

class TestComponent extends Component {
  state = {};

  render() {
    return (
      <div className=\\"TestComponent\\"></div>
    );
  }
}

export default TestComponent;
",
]
`;

exports[`generate should change css extension 1`] = `
".T {}
"
`;

exports[`generate should fail 1`] = `"src is not defined"`;

exports[`generate should generate a component with default config 1`] = `"TestComponent"`;

exports[`generate should generate a component with default config 2`] = `
Array [
  ".TestComponent {}
",
  "export { default } from './TestComponent'
",
  "import React from 'react'
import ReactDOM from 'react-dom'
import TestComponent from './TestComponent'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<TestComponent />, div)
})
",
  "import React, { Component } from 'react'
import './TestComponent.css'

class TestComponent extends Component {
  state = {}

  render() {
    return (
      <div className=\\"TestComponent\\"></div>
    )
  }
}

export default TestComponent
",
]
`;

exports[`generate should generate functional components 1`] = `
"import React from 'react'
import './TestComponent.css'

const TestComponent = ({}) => (
  <div className=\\"TestComponent\\"></div>
)

export default TestComponent
"
`;

exports[`generate should use flow type 1`] = `
Array [
  "// @flow

export { default } from './T'
",
  "// @flow

import React from 'react'
import ReactDOM from 'react-dom'
import T from './T'

it('renders without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<T />, div)
})
",
  "// @flow

import React, { Component } from 'react'
import './T.css'

class T extends Component {
  state = {}

  render() {
    return (
      <div className=\\"T\\"></div>
    )
  }
}

export default T
",
]
`;
